### PR TITLE
MINOR: Add `KafkaAdminClient.getListOffsetsCalls` benchmark

### PR DESCRIPTION
**Pull request by jeffkbkim**
new benchmark to test KafkaAdminClient.getListOffsetsCalls
originated from https://github.com/apache/kafka/pull/10940

### Committer Checklist (excluded from commit message)
- [ ] Verify design and implementation 
- [ ] Verify test coverage and CI build status
- [ ] Verify documentation (including upgrade notes)

### CODE REVIEW

**Comment by jeffkbkim** [comment]
ran benchmark with warm up = 5, iteration = 5

With changes:
```
Benchmark                                             (partitionCount)  (topicCount)  Mode  Cnt       Score       Error  Units
GetListOffsetsCallsBenchmark.testGetListOffsetsCalls               100             1  avgt    5      40.040 ±     3.506  us/op
GetListOffsetsCallsBenchmark.testGetListOffsetsCalls               100            10  avgt    5     474.825 ±    95.547  us/op
GetListOffsetsCallsBenchmark.testGetListOffsetsCalls              1000             1  avgt    5     429.767 ±    60.605  us/op
GetListOffsetsCallsBenchmark.testGetListOffsetsCalls              1000            10  avgt    5    5069.417 ±   662.226  us/op
GetListOffsetsCallsBenchmark.testGetListOffsetsCalls             10000             1  avgt    5    6000.379 ±   235.654  us/op
GetListOffsetsCallsBenchmark.testGetListOffsetsCalls             10000            10  avgt    5  121240.173 ± 49894.963  us/op
```

Without changes:
```
GetListOffsetsCallsBenchmark.testGetListOffsetsCalls partition=100,    topic=1: Iteration   5: 4860.989 us/op
GetListOffsetsCallsBenchmark.testGetListOffsetsCalls partition=100,   topic=10: Iteration   1: 424064.225 us/op
GetListOffsetsCallsBenchmark.testGetListOffsetsCalls partition=1000,   topic=1: Iteration   5: 452289.400 us/op
GetListOffsetsCallsBenchmark.testGetListOffsetsCalls partition=1000,  topic=10: Iteration   2: 49306777.392 us/op
GetListOffsetsCallsBenchmark.testGetListOffsetsCalls partition=10000,  topic=1: Iteration   5: 53940393.545 us/op
GetListOffsetsCallsBenchmark.testGetListOffsetsCalls partition=10000, topic=10: Warm-up Itr 1: 14664107350.721 us/op
```
Running the benchmark pre-https://github.com/apache/kafka/pull/10940 took too long to finish. included the median result for each topic-partition pair. case for partition=10000, topic=10 took too long during warm up (each take +4 hours).

File: clients/src/main/java/org/apache/kafka/clients/admin/KafkaAdminClient.java [hunk 1]
```diff
@@ -3677,7 +3677,8 @@ public ListOffsetsResult listOffsets(Map<TopicPartition, OffsetSpec> topicPartit
         return new ListOffsetsResult(new HashMap<>(futures));
     }
 
-    private List<Call> getListOffsetsCalls(MetadataOperationContext<ListOffsetsResultInfo, ListOffsetsOptions> context,
+    // visible for benchmark
+    public List<Call> getListOffsetsCalls(MetadataOperationContext<ListOffsetsResultInfo, ListOffsetsOptions> context,
```
Comment (dajac) [inline clients/src/main/java/org/apache/kafka/clients/admin/KafkaAdminClient.java]: I am a bit concerned by making the method public here. `KafkaAdminClient` is directly used by our users so there is a risk that they start calling this method. I wonder if we could at least keep it package private.

**Comment by dajac** [comment]
@jeffkbkim We could add a `getListOffsetsCalls` method to `AdminClientTestUtils`, which is in the same package as the `KafkaAdminClient`, to access the package private method from the benchmark.

File: clients/src/main/java/org/apache/kafka/clients/admin/KafkaAdminClient.java [hunk 2]
```diff
@@ -3677,7 +3677,8 @@ public ListOffsetsResult listOffsets(Map<TopicPartition, OffsetSpec> topicPartit
         return new ListOffsetsResult(new HashMap<>(futures));
     }
 
-    private List<Call> getListOffsetsCalls(MetadataOperationContext<ListOffsetsResultInfo, ListOffsetsOptions> context,
+    // visible for benchmark
+    List<Call> getListOffsetsCalls(MetadataOperationContext<ListOffsetsResultInfo, ListOffsetsOptions> context,
```
Comment (dajac) [inline clients/src/main/java/org/apache/kafka/clients/admin/KafkaAdminClient.java:3681]: nit: Could we realign the arguments?

**Comment by jeffkbkim** [comment]
@dajac thank you for the review and suggestion! i have addressed your comments.

File: clients/src/test/java/org/apache/kafka/clients/admin/AdminClientTestUtils.java [hunk 3]
```diff
@@ -102,4 +104,15 @@ public static DescribeTopicsResult describeTopicsResult(Map<String, TopicDescrip
     public static ListConsumerGroupOffsetsResult listConsumerGroupOffsetsResult(Map<TopicPartition, OffsetAndMetadata> offsets) {
         return new ListConsumerGroupOffsetsResult(KafkaFuture.completedFuture(offsets));
     }
+
+    /**
+     * Used for benchmark. KafkaAdminClient.getListOffsetsCalls is only accessible
+     * from within the admin package.
+     */
+    public static List<KafkaAdminClient.Call> getListOffsetsCalls(KafkaAdminClient adminClient,
```
Comment (dajac) [inline clients/src/test/java/org/apache/kafka/clients/admin/AdminClientTestUtils.java:112]: Could we align the arguments?

**Review by dajac — APPROVED** [review]
LGTM

**Pull request merged**